pkg/webhook/validate_service.go

```diff
@@ -56,5 +59,14 @@ func ValidateRevisionTemplate(ctx context.Context, uns *unstructured.Unstructure
 	if templ == nil || templ == (&v1.RevisionTemplateSpec{}) {
 		return nil // Don't need to validate empty templates
 	}
+
+	if apis.IsInUpdate(ctx) {
+		if uns, err := runtime.DefaultUnstructuredConverter.ToUnstructured(apis.GetBaseline(ctx)); err == nil {
+			if oldVal, found, err := unstructured.NestedFieldNoCopy(uns, "spec", "template"); found && err == nil &&
```

Comment:
I'd rearrange `err == nil && found && equal` since that's the order you probably compare when you think of it. Also, since you're ignoring the error (in a sense) you can just check `found && equal`, I think. It should not be true if the field extraction failed.